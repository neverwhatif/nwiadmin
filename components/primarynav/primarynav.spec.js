import React from 'react';
import { shallow } from 'enzyme';

import { PrimaryNavComponent } from './index';

const mockLocation = {
    pathname: '/lorem',
    search: '',
};

describe('PrimaryNavComponent', () => {
    it('should render', () => {
        expect(shallow(<PrimaryNavComponent location={mockLocation} />)).toMatchSnapshot();
    });
});
