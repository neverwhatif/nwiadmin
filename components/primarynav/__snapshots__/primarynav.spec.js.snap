// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryNavComponent should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrimaryNavComponent
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <PrimaryNavItem
            isActive={false}
            isDisabled={false}
            path="/members"
            title="Members"
          />
        </li>,
        <li>
          <PrimaryNavItem
            isActive={false}
            isDisabled={false}
            path="/projects"
            title="Projects"
          />
        </li>,
        <li>
          <PrimaryNavItem
            isActive={false}
            isDisabled={false}
            path="/investments"
            title="Investments"
          />
        </li>,
        <li>
          <PrimaryNavItem
            isActive={false}
            isDisabled={false}
            path="/shareregister"
            title="Share Register"
          />
        </li>,
        <li>
          <PrimaryNavItem
            isActive={false}
            isDisabled={false}
            path="/reports"
            title="Reports"
          />
        </li>,
        <li>
          <PrimaryNavItem
            isActive={false}
            isDisabled={true}
            path="/system"
            title="System"
          />
        </li>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <PrimaryNavItem
            isActive={false}
            isDisabled={false}
            path="/members"
            title="Members"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "isActive": false,
            "isDisabled": false,
            "path": "/members",
            "title": "Members",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <PrimaryNavItem
            isActive={false}
            isDisabled={false}
            path="/projects"
            title="Projects"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "isActive": false,
            "isDisabled": false,
            "path": "/projects",
            "title": "Projects",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <PrimaryNavItem
            isActive={false}
            isDisabled={false}
            path="/investments"
            title="Investments"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "isActive": false,
            "isDisabled": false,
            "path": "/investments",
            "title": "Investments",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": <PrimaryNavItem
            isActive={false}
            isDisabled={false}
            path="/shareregister"
            title="Share Register"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "isActive": false,
            "isDisabled": false,
            "path": "/shareregister",
            "title": "Share Register",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": <PrimaryNavItem
            isActive={false}
            isDisabled={false}
            path="/reports"
            title="Reports"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "isActive": false,
            "isDisabled": false,
            "path": "/reports",
            "title": "Reports",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": <PrimaryNavItem
            isActive={false}
            isDisabled={true}
            path="/system"
            title="System"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "isActive": false,
            "isDisabled": true,
            "path": "/system",
            "title": "System",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <PrimaryNavItem
              isActive={false}
              isDisabled={false}
              path="/members"
              title="Members"
            />
          </li>,
          <li>
            <PrimaryNavItem
              isActive={false}
              isDisabled={false}
              path="/projects"
              title="Projects"
            />
          </li>,
          <li>
            <PrimaryNavItem
              isActive={false}
              isDisabled={false}
              path="/investments"
              title="Investments"
            />
          </li>,
          <li>
            <PrimaryNavItem
              isActive={false}
              isDisabled={false}
              path="/shareregister"
              title="Share Register"
            />
          </li>,
          <li>
            <PrimaryNavItem
              isActive={false}
              isDisabled={false}
              path="/reports"
              title="Reports"
            />
          </li>,
          <li>
            <PrimaryNavItem
              isActive={false}
              isDisabled={true}
              path="/system"
              title="System"
            />
          </li>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <PrimaryNavItem
              isActive={false}
              isDisabled={false}
              path="/members"
              title="Members"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "isActive": false,
              "isDisabled": false,
              "path": "/members",
              "title": "Members",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <PrimaryNavItem
              isActive={false}
              isDisabled={false}
              path="/projects"
              title="Projects"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "isActive": false,
              "isDisabled": false,
              "path": "/projects",
              "title": "Projects",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <PrimaryNavItem
              isActive={false}
              isDisabled={false}
              path="/investments"
              title="Investments"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "isActive": false,
              "isDisabled": false,
              "path": "/investments",
              "title": "Investments",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <PrimaryNavItem
              isActive={false}
              isDisabled={false}
              path="/shareregister"
              title="Share Register"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "isActive": false,
              "isDisabled": false,
              "path": "/shareregister",
              "title": "Share Register",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": <PrimaryNavItem
              isActive={false}
              isDisabled={false}
              path="/reports"
              title="Reports"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "4",
            "nodeType": "function",
            "props": Object {
              "isActive": false,
              "isDisabled": false,
              "path": "/reports",
              "title": "Reports",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": <PrimaryNavItem
              isActive={false}
              isDisabled={true}
              path="/system"
              title="System"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "5",
            "nodeType": "function",
            "props": Object {
              "isActive": false,
              "isDisabled": true,
              "path": "/system",
              "title": "System",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
